modules/top-level/files/submodule.nix

```diff
@@ -22,6 +23,10 @@
       path = lib.mkDefault name;
       type = lib.mkDefault (if lib.hasSuffix ".vim" name then "vim" else "lua");
       # No need to use mkDerivedConfig; this option is readOnly.
-      plugin = helpers.writeLua derivationName config.content;
+      plugin =
+        let
+          writeContent = if config.type == "lua" then helpers.writeLua else pkgs.writeText;
```

Comment:
I think it's more readable with this in the same let block as `derivationName`.

Also: in #1889 I named the function `writer`, although that's not really important. `writeContent` is fine, just a little verbose.

Reply:
Maybe I'm wrong, but isn't a verb better in that case? I named it by analogy to `writeLua` and `writeText`.